buildkite/resource_cluster_queue.go

```diff
@@ -95,6 +129,52 @@ func (clusterQueueResource) Schema(ctx context.Context, req resource.SchemaReque
 				MarkdownDescription: "The dispatch state of a cluster queue.",
 				Default:             booldefault.StaticBool(false),
 			},
+			"hosted_agents": resource_schema.SingleNestedAttribute{
+				Optional:            true,
+				MarkdownDescription: "Control the settings for the Buildkite hosted agents.",
+				Validators: []validator.Object{
+					&hostedAgentValidator{},
+				},
+				Attributes: map[string]resource_schema.Attribute{
+					"mac": resource_schema.SingleNestedAttribute{
+						Optional: true,
+						Attributes: map[string]resource_schema.Attribute{
+							"xcode_version": resource_schema.StringAttribute{
+								Optional: true,
+								Computed: true,
+							},
+						},
+					},
+					"linux": resource_schema.SingleNestedAttribute{
+						Optional: true,
+						Attributes: map[string]resource_schema.Attribute{
+							"image_agent_ref": resource_schema.StringAttribute{
```

Comment:
The naming of this field in the API is `agent_image_ref`, we should be consistent with the naming if we're doing something similar.

Reply:
I've updated this attribute now to keep the naming consistent